frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsForm.tsx

```diff
@@ -0,0 +1,193 @@
+    MembersAPI.getAllMembers().then((members) => setMembersList(members));
+    CoffeeChatAPI.getCoffeeChatsByUser(userInfo).then((coffeeChats) => {
+      setApprovedChats(coffeeChats.filter((chat) => chat.status === 'approved'));
+      setPendingChats(coffeeChats.filter((chat) => chat.status === 'pending'));
+      setRejectedChats(coffeeChats.filter((chat) => chat.status === 'rejected'));
+      setIsChatLoading(false);
+    });
+    CoffeeChatAPI.getCoffeeChatBingoBoard().then((board) => setBingoBoard(board));
+  }, [userInfo]);
+
+  const coffeeChatExists = (): boolean =>
+    approvedChats.some((chat) => chat.otherMember.netid === member?.netid) ||
+    pendingChats.some((chat) => chat.otherMember.netid === member?.netid);
+
+  const submitCoffeeChat = async () => {
+    if (!member) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Member Selected',
+        contentMsg: 'Please select a team member!'
+      });
+    } else if (!slackLink) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Slack Link Entered',
+        contentMsg: 'Please enter a link to your Slack post in #coffee-chats!'
+      });
+    } else if (!category) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Category Selected',
+        contentMsg: 'Please select a category!'
+      });
+    } else if (member.netid === userInfo.netid) {
+      Emitters.generalError.emit({
+        headerMsg: 'Cannot Coffee Chat Yourself',
+        contentMsg: 'Please submit a coffee chat with another member!'
+      });
+    } else if (coffeeChatExists()) {
+      Emitters.generalError.emit({
+        headerMsg: 'Coffee Chat Exists',
+        contentMsg: 'Please submit a new coffee chat!'
+      });
+    } else {
+      const newCoffeeChat: CoffeeChat = {
+        uuid: '',
+        submitter: userInfo,
+        otherMember: member,
+        slackLink,
+        category,
+        status: 'pending' as Status,
+        date: new Date().getTime()
+      };
+
+      CoffeeChatAPI.createCoffeeChat(newCoffeeChat).then(() => {
+        setPendingChats((pending) => [...pending, newCoffeeChat]);
+      });
+      Emitters.generalSuccess.emit({
+        headerMsg: 'Coffee Chat submitted!',
+        contentMsg: `The leads were notified of your submission, and your coffee chat will be approved soon!`
+      });
+      setMember(undefined);
+      setCategory('');
+      setSlackLink('');
+    }
+  };
+
+  return (
+    <div>
+      <Form className={styles.form_style}>
+        <h1>Submit Coffee Chats</h1>
+        <p>
+          Submit your coffee chats to earn credit for the Coffee Chat Bingo! Fill out this form
+          after each chat to receive your credit.
+        </p>
+        <div className={styles.inline}>
+          <label className={styles.bold}>
+            Select a team member: <span className={styles.red_color}>*</span>
+          </label>
+          <div className={styles.center_and_flex}>
+            {membersList ? (
+              <Dropdown
```

Comment:
I think still for the search bar, I can only search for people in the search bar by NetID